packages/react-reconciler/src/ReactChildFiber.old.js

```diff
@@ -1385,14 +1388,16 @@ function ChildReconciler(shouldTrackSideEffects) {
         // Intentionally fall through to the next case, which handles both
         // functions and classes
         // eslint-disable-next-lined no-fallthrough
-        case FunctionComponent: {
-          const Component = returnFiber.type;
+        case FunctionComponent:
+        case MemoComponent:
+        case SimpleMemoComponent:
+        case ForwardRef: {
```

Comment:
Block, too

Reply:
I don't think you need MemoComponent, since that one always wraps around another fiber. SimpleMemoComponent  should be sufficient.

Reply:
Oh right. I was wondering if I was forgetting one.

Too many component types 😆

Reply:
Hm. I can just add the case, but I'm not sure how to write a test that hits this code path for `Block`.